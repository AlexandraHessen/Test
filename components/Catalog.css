.Catalog{
    display: flex;
    justify-content: space-between;
}
.Pagination{
    background-color: #fff;
    display: flex;
    padding: 0 0 30px 30px;
    flex-grow: 1;
    flex-shrink: 1;
    justify-content: flex-start;
} 

.Pagination > li {
    display: inline;
}

.PaginationLink{
    position: relative;
    float: left;
    padding: 7px 14px;
    margin-left: -1px;
    font-size: 14px;
    font-weight: bold;
    letter-spacing: 0.3px;
    color: #969696;
    text-decoration: none;
    background-color: #f8f8f8;
    border: 1px solid #f8f8f8;
    border-radius: 0;
    margin: 0 1px;
}

.ActivePaginationLink{
    background: #fff;
    color: #357A10;
    border: solid 1px #357A10;
}

.NullResults{
    color: #363636;
    padding: 10px;
    background-color: #fff;
    padding: 15px;
    min-height: 700px;
    width: 100%;
}

.Filter{
    /* background: rgba(255, 255, 255, 0.3); */
    /* list-style-type: none; */
    /* max-width: 300px; */
    margin-right: 20px;
    /* width: 25%; */
}

.LabelFilterPrice{
    font-family: 'PT Sans',sans-serif;
    color: #fff;
    font-weight: bold;
}

.InputFilterPrice{
    border: 1px solid #fff;
    border-radius: 3px;
    margin: 10px 0;
}

.FilterButtons{
    border: none;
    padding: 4px;
    color: #fff;
    font-size: 0.75em;
    background: #57ad26;
    border-radius: 5px;
    cursor: pointer;
    margin-right: 5px;
}

input[type="button"].FilterButtons:last-of-type{
    margin-right: 0;
}

@media all and (max-width: 980px){
    .Catalog{
        flex-direction: column;
    }

    .Filter{
        text-align: end;
        margin-bottom: 10px;
    }

    .LabelFilterPrice{
        margin-right: 10px;
    }

    .InputFilterPrice{
        margin: 0 0 10px;
        text-align: end;
    }
}

@media all and (max-width: 500px){
    .InputFilterPrice{
        width: 20%;
    }
}