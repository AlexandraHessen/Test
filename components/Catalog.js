import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import { NavLink } from 'react-router-dom';

import PlantCard from '../components/PlantCard';

class Catalog extends React.PureComponent{
    static propTypes={
        plantsArr: PropTypes.array.isRequired,
        // pageNumber: PropTypes.string.isRequired
    }

    state ={
        countOfPlantsOnOnePage: 20
    }

    render(){
        let plantsArr=this.props.plantsArr;
        let countOfAllPlants=plantsArr.length; //общее кол-во всех элементов 
        let pageLinksArr=[];
        let countOfAllPages=Math.ceil(countOfAllPlants/this.state.elemOnPage); //общее кол-во  страниц
        for (let i=1; i++; i<countOfAllPages){ //строим сами теги страниц
            pageLinksArr.push(<NavLink to={`/catalog/${i}`}>{i}</NavLink>)
        }

        let pageNumber=parseInt(this.props.pageNumber) //номер страницы

        let startIndex = (pageNumber-1) * this.state.countOfPlantsOnOnePage; // индекс первого элемента на странице (с какого?) номер страницы минус 1 * количество  элементов на странице
        let endIndex = startIndex + countOfPlantsOnOnePage //индекс последнего элемента на странице (по какой?) индекс первого товара на странице + кол-во товаров на странице

        let objsPlantOnOnePage = plantsArr.slice(startIndex, endIndex) // элементы на одной странице ( общее количество делим slice от и до НЕ ВКЛЮЧИТЕЛЬНО)


        // quantityPlantOnPage
        // objPlantOnOnePage 

        // какие элементы отображать на 1 странице 


        let plantsArrCode=this.props.plantsArr.map( plant=>
            <PlantCard key={plant.code} info={plant} />
            )

        return(
        <div>
            <div className="CatalogPage">
                {plantsArrCode}
            </div>

            {/* <div>
               { pageLinksArr}
            </div> */}
        </div>

        )

    }

}

export default Catalog 