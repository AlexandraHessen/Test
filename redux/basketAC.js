// ---------------- ActionCreator ----------------//
const ADD_PRODUCT = "ADD_PRODUCT"
const DEL_PRODUCT = "DEL_PRODUCT"
const CLEAR_BASKET = "CLEAR_BASKET" //пользователь сам очищает корзину при желании

//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator

const add_product = function(productId, objProductInfo) {
    return {
      type: ADD_PRODUCT,
      productId: productId,
      objProductInfo: objProductInfo
    };
  }
  
  const del_product = function(productId) {
    return {
      type: DEL_PRODUCT,
      productId: productId,
    };
  }

  const clear_basket = function() {
    return {
      type: CLEAR_BASKET
    };
  }

export {
    add_product, ADD_PRODUCT,
    del_product, DEL_PRODUCT,
    clear_basket, CLEAR_BASKET,
  }