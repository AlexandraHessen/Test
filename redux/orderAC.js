// ---------------- ActionCreator ----------------//
const ADD_ORDER = "ADD_ORDER"

//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator

const add_order = function(objCustomerInfo, objOrderInfo) {
    return {
      type: ADD_ORDER,
      objCustomerInfo: objCustomerInfo,
      objOrderInfo: objOrderInfo
    };
  }

export {
    add_order, ADD_ORDER,
  }