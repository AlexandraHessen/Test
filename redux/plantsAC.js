// ---------------- ActionCreator ----------------//
const PLANTS_LOADING='PLANTS_LOADING';
const PLANTS_ERROR='PLANTS_ERROR';
const PLANTS_SET='PLANTS_SET';
//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator
const plantsLoadingAC=function() {
  return {
    type: PLANTS_LOADING,
  };
}

const plantsErrorAC=function() {
  return {
    type: PLANTS_ERROR,
  };
}

const plantsSetAC=function(plants) {
  return {
    type: PLANTS_SET,
    plants:plants,
  };
}

export {
  plantsLoadingAC,PLANTS_LOADING,
  plantsErrorAC,PLANTS_ERROR,
  plantsSetAC,PLANTS_SET,
}
