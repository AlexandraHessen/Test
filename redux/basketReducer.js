// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { ADD_PRODUCT, DEL_PRODUCT, CLEAR_BASKET } from './basketAC'; //название action type


const initState={ // начальный state
    productsInBasket: {}
    //КЛЮЧИ productId, objProductInfo
    // 1: {code: 1, name: 'Замиокулькас'}

//   status: 0, // 0 - ничего не началось, 1 - идёт загрузка, 2 - была ошибка, 3 - данные загружены
//   data: null, //даные которые мы загрузили по сети

}

function basketReducer(state=initState,action) {
  switch (action.type) {

    case ADD_PRODUCT: {
      console.log('action:',action);
      console.log('state до обработки редьюсером:',state);
      let newState={...state, //старый state
        productsInBasket:{...state.productsInBasket, //заменить старые продукты на теже +
          // ПРИЛЕТИТ какой продукт (productid) изменить и насколько(addvalue)
          [action.productId]:action.objProductInfo //+
          //[action.counterid] - В нужном продукте 
          // state.productsInBasket[action.productid] - его старое значение 
          // action.addvalue - изменить на прилетевшее
        }
      };
      console.log('state после обработки редьюсером:',newState);
      return newState;
    }

    case DEL_PRODUCT: {
      let newState = {...state};
      delete newState.productsInBasket[action.productId];
      return newState;
    }

    case CLEAR_BASKET:{
      let newState = {...state};
      newState.productsInBasket={};
      return newState;
    }

    // case PLANTS_ERROR: {
    //   let newState={
    //     status:2,
    //     data:null,
    //   };
    //   return newState;
    // }

    // case PLANTS_SET: {
    //   let newState={
    //     status:3,
    //     data:action.plants, //загруженные данные
    //   };
    //   return newState;
    // }
    
//обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default basketReducer;
