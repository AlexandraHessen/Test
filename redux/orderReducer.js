// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { ADD_ORDER } from './orderAC'; //название action type

const initState={ // начальный state
    orders: []
}

function orderReducer(state=initState,action) {
  switch (action.type) {
    case ADD_ORDER: {
      let newOrder=[...state.orders]
      newOrder.push( { objCustomerInfo: action.objCustomerInfo, products: action.objOrderInfo } );
      let newState={...state, orders: newOrder};

      return newState;
    }

              //обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default orderReducer;