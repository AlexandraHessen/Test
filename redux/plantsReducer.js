// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { PLANTS_LOADING, PLANTS_ERROR, PLANTS_SET } from './plantsAC'; //название action type


const initState={ // начальный state

  status: 0, // 0 - ничего не началось, 1 - идёт загрузка, 2 - была ошибка, 3 - данные загружены
  data: null, //даные которые мы загрузили по сети

}

function plantsReducer(state=initState,action) {
  switch (action.type) {

    case PLANTS_LOADING: {
      let newState={
        status:1,
        data:null,
      };
      return newState;
    }

    case PLANTS_ERROR: {
      let newState={
        status:2,
        data:null,
      };
      return newState;
    }

    case PLANTS_SET: {
      let newState={
        status:3,
        data:action.plants, //загруженные данные
      };
      return newState;
    }
    
//обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default plantsReducer;
