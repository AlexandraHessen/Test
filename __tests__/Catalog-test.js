"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import { createStore } from 'redux';
import combinedReducer from '../redux/reducers.js';
import { Provider } from 'react-redux';

import Catalog from '../components/Basket';
                    //импортируем файл который будем тестировать

let plantsArr = [
    {
		"code": 1,
		"name": "Замиокулькас",
		"category": "deciduous",
		"price": 265,
		"description": "Полив по мере полного просыхания земляного кома, раз в 10-15 дней, в опрыскивании не нуждается. Чрезмерный полив замиокулькасов может приводить к пожелтению листьев.",
		"imgUrl": "/img/img_9241_1.jpg"
	},
	{
		"code": 2,
		"name": "Филодендрон Malay Gold",
		"category": "deciduous",
		"price": 95,
		"description": "Кашпо на фото не входит в стоимость, растение продается в пластиковом техническом горшочке",
		"imgUrl": "/img/img_6587.jpg"
	},
	{
		"code": 3,
		"name": "Монстера Monkey Leaf",
		"category": "deciduous",
		"price": 385,
		"description": "Стоимость указана за одно растение в техническом горшке, кашпо на фото не входит в стоимость",
		"imgUrl": "/img/img_6129.jpg"
	},
	{
		"code": 4,
		"name": "Аспелениум",
		"category": "deciduous",
		"price": 58,
		"description": "Стоимость указана за одно растение в техническом горшке, кашпо на фото не входит в стоимость",
		"imgUrl": "/img/img_6162_2.jpg"
	},
	{
		"code": 5,
		"name": "Кхенанта Amagris",
        "category": "deciduous",
		"price": 45,
		"description": "Стоимость указана за растение в техническом пластиковом горшке, кашпо на фото не входит в стоимость.",
		"imgUrl": "/img/img_6053.jpg"
	},
	{
		"code": 6,
		"name": "Кордилина",
        "category": "deciduous",
		"price": 32,
		"description": "Кашпо не входит в стоимость",
		"imgUrl": "/img/img_5677.jpg"
	},
	{
		"code": 7,
		"name": "Бонсай Гинсенг в керамике",
        "category": "deciduous",
		"price": 105,
		"description": "Кашпо входит в стоимость",
		"imgUrl": "/img/img_3604_1.jpg"
	},
	{
		"code": 8,
		"name": "Замиокулкас",
        "category": "deciduous",
		"price": 47,
		"description": "Стоимость указана за растение в пластиковом техническом горшке, керамическое кашпо не входит в стоимость.",
		"imgUrl": "/img/img_4001.jpg"
	},
	{
		"code": 9,
		"name": "Шефлера",
        "category": "deciduous",
		"price": 315,
		"description": "Стоимость указана за растение в пластиковом горшке.",
		"imgUrl": "/img/img_0019.jpg"
	},
	{
		"code": 10,
		"name": "Фикус микс",
        "category": "deciduous",
		"price": 45,
		"description": "Растения представлены миксом, наличие нужного сорта уточняйте у специалистов салона. Стоимость указана за растение в пластиковом горшке. Подобрать кашпо или заказать вариант, представленный на фото, можно в салоне.",
		"imgUrl": "/img/img_3543.jpg"
	},
	{
		"code": 11,
		"name": "Бонсай Фикус Гинсенг",
        "category": "deciduous",
		"price": 21,
		"description": "Керамическое кашпо не входит в стоимость, растение продается в техническом горшочке",
		"imgUrl": "/img/img_4948.jpg"
	},
	{
		"code": 12,
		"name": "Пуансетия",
        "category": "deciduous",
		"price": 25,
		"description": "Стоимость указана за растение в техническом пластиковом горшке без учета керамического кашпо.",
		"imgUrl": "/img/img_6779_2.jpg"
	},
    {
		"code": 13,
		"name": "Сциндапсус Argyraeus",
        "category": "deciduous",
		"price": 25,
		"description": "Сциндапсус предпочитает расти в тени, либо полутени. Поливать растение требуется только после того, как 1/3 часть грунта просохнет. ",
		"imgUrl": "/img/img_0204.jpg"
	},
	{
		"code": 14,
		"name": "Мирт communis",
        "category": "deciduous",
		"price": 85,
		"description": "Главное для мирта – постоянный равномерный полив без перерывов и поддержание повышенной влажности воздуха. Почва всегда должна быть увлажнена, поэтому в зависимости от условий в комнате мирт в среднем стоит поливать ежедневно или через день. ",
		"imgUrl": "/img/img_6697_1.jpg"
	},
	{
		"code": 15,
		"name": "Маранта Керховена",
        "category": "deciduous",
		"price": 23,
		"description": "Стоимость указана за растение в пластиковом горшке. Подобрать кашпо или заказать вариант, представленный на фото, можно в салоне.",
		"imgUrl": "/img/img_9319.jpg"
	}
]

let pageNumber = 2;
let store=createStore(combinedReducer);

test('Catalog', () => {
                            // создаём тестовую версию компонента который будем тестировать
  const component = renderer.create(
  <Provider store={store}>
    <BrowserRouter>
      <Catalog plantsArr = {plantsArr} pageNumber = {pageNumber}/>
    </BrowserRouter>
  </Provider>

  );

                            // получаем снэпшот (HTML-снимок) компонента для сверки, чтобы вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

    
});
