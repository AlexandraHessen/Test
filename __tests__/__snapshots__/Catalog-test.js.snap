// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog 1`] = `
<div>
  <table
    className="TableBasket"
  >
    <tbody>
      <tr
        className="TableHeader"
      >
        <th>
          Товар
        </th>
        <th />
        <th
          className="priceTh"
        >
          Цена
        </th>
        <th />
      </tr>
    </tbody>
  </table>
  <div
    className="Total"
  >
    Итого:
    <div
      className="SumProductsInBasket"
    >
      0
       руб.
    </div>
  </div>
  <div
    className="ActionsBasket"
  >
    <input
      className="BasketButtons"
      onClick={[Function]}
      type="button"
      value="Очистить корзину"
    />
    <input
      className="BasketButtons"
      onClick={[Function]}
      type="button"
      value="Оформить заказ"
    />
  </div>
</div>
`;
